### PR TITLE
C++: Fix `asUninitialized`

### PR DESCRIPTION
**Pull request by MathiasVP**
We've seen a number of weird results when using the IR version of the `UninitializedNode` class. This should hopefully fix the weirdness.

### CODE REVIEW

File: cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp [hunk 1]
```diff
@@ -423,17 +423,17 @@ void intPointerSourceCaller() {
...
+
+
+
+
 void intPointerSourceCaller2() {
   int local[1];
   intPointerSource(local);
-  sink(local); // $ ast=433:7 ast=434:20 ir=433:7 ir=434:20 ir
-  sink(*local); // $ ast=433:7 ast=434:20 ir=433:7 ir=434:20
+  sink(local); // $ ast=434:20 ir SPURIOUS: ast=433:7
```
Comment (jketema) [inline cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp:435]: I'm not expecting flow here, as discussed at https://github.com/github/codeql/pull/11254#discussion_r1028503418.

Comment (MathiasVP) [reply cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp:435]: True. This PR is removing _some_ of the flow, but there's still some that need to be sorted out.

Comment (jketema) [reply cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp:435]: Got it. Thanks.

Comment (MathiasVP) [reply cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp:435]: Oh, wait. We're getting flow here because of https://github.com/github/codeql/blob/mathiasvp/replace-ast-with-ir-use-usedataflow/cpp/ql/test/library-tests/dataflow/dataflow-tests/test.ql#L84 (and indeed *local is tainted).

File: cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp [hunk 2]
```diff
@@ -594,7 +594,7 @@ void test_indirect_flow_to_array() {
   int* p = indirect_source();
   int* xs[2];
   xs[0] = p;
-  sink(*xs[0]); // $ ir=594:12 MISSING: ast SPURIOUS: ir=595:8
+  sink(*xs[0]); // $ ir MISSING: ast
```
Comment (jketema) [inline cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp:597]: I still find this hard to follow. Which flow remains here? `ir=594:12` or `ir=595:8`? I actually suspect `ir=595:8`?

Comment (MathiasVP) [reply cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp:597]: Yeah, I agree. It's kinda difficult to see what's going on here when we cannot label the single source. I'll check this and write out which source we're actually catching here.

Comment (MathiasVP) [reply cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp:597]: FWIW, we're actually keeping the `ir=594:12` result 🎉 
Screenshot 2023-03-08 at 09 04 27

Comment (MathiasVP) [reply cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp]: I've clarified the result in https://github.com/github/codeql/pull/12432/commits/263b2082828a36b9a70d415eb0b0ce0b22adb136. Obviously, it would be nice to have an enforced annotation to capture this, but that's not possible (without introducing another source of flow that hits the sink).

File: cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp (hunk 1 above)
Comment (jketema) [reply cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp:435]: Ah ok. Then this looks fine.

**Review by jketema — APPROVED** [review]
LGTM.

**Comment by MathiasVP** [comment]
CI failures are unrelated. 🤠 merging!
